File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 04/04/2026, 23:37:00
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int buy = prices[0];
        int ans = 0;
        for(int i=1;i<n;i++) {
            if(prices[i] < buy) {
                buy = prices[i];
            } else {
                ans = max(ans, prices[i]-buy);
            }
        }
        return ans;
    }
};

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 04/04/2026, 23:36:56
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0];
        int ans = 0;
        for(int i=1;i<prices.length;i++) {
            if(min > prices[i]) min = prices[i];
            else if(prices[i]-min > ans) ans = prices[i]-min;
        }
        return ans;
    }
}